packages/itwin/measure-tools/src/tools/MeasureLocationTool.ts

```diff
@@ -152,12 +155,20 @@ MeasureLocationToolModel
     if (!this._enableSheetMeasurements || !ev.viewport?.view.isSheetView())
       return true;
 
-    if (!SheetMeasurementsHelper.checkIfAllowedDrawingType(ev, [SheetMeasurementsHelper.DrawingType.CrossSection, SheetMeasurementsHelper.DrawingType.Plan]))
+    if (!SheetMeasurementsHelper.checkIfAllowedDrawingType(ev.viewport, ev.point, MeasureLocationTool._allowedDrawingTypes))
       return false;
 
     return true;
   }
 
+  public override async getToolTip(hit: HitDetail): Promise<HTMLElement | string> {
+    if (!this._enableSheetMeasurements || SheetMeasurementsHelper.checkIfAllowedDrawingType(hit.viewport, hit.hitPoint, MeasureLocationTool._allowedDrawingTypes)) {
+      return super.getToolTip(hit);
+    } else {
+      return SheetMeasurementsHelper.getToolTipText(MeasureLocationTool._allowedDrawingTypes);
```

Comment:
`SheetMeasurementsHelper.getToolTipText` should be named to reflect it's specifically getting an error text. Otherwise, to reduce the boilerplate code these tool getToolTips can become:

```
if (!this._enableSheetMeasurements)
  return super.getToolTip(hit);

return SheetMeasurementsHelper.getToolTipText(hit, Tool._allowedDrawingTypes, async () => super.getToolTip(hit));
```

e.g. The helper checks if it's allowed, calls the function that is passed in, otherwise returns the error text.

Also, this feels like it could be in the base measurement tool class. And the list of allowed drawing types either like how subclass tools override static properties, or it be an instance-based method that is overridden that returns the array.

Reply:
@bsy-nicholasw I think I've implemented your suggestion, please verify when you have time.

Reply:
Yeah, the tool changes look good. I made some other comments about the helper